src/fprime/fbuild/gcovr.py

```diff
@@ -227,60 +103,108 @@ def execute(
                 file=sys.stderr,
             )
             return
-        ac_temporary_path = _get_ac_directory(builder, context, self.scope)
-        coverage_output_dir = context / "coverage"
+        build_cache_resolved = Path(builder.build_dir).absolute().resolve()
+
+        _, pass_through_args, options = args
+        include_all_ac = options["--all-ac"]
+        include_comp_ac = include_all_ac or options["--comp-ac"]
+        include_port_ac = include_all_ac or options["--port-ac"]
+        include_type_ac = include_all_ac or options["--type-ac"]
+
+        exclusion_filter_bases = [
+            None if include_comp_ac else ".*ComponentAc.[ch]pp",
+            None if include_port_ac else ".*PortAc.[ch]pp",
+            None if include_type_ac else ".*SerializableAc.[ch]pp",
+            None if include_type_ac else ".*ArrayAc.[ch]pp",
+            None if include_type_ac else ".*EnumAc.[ch]pp",
+        ]
+        exclusion_filter_bases = filter(
+            lambda item: item is not None, exclusion_filter_bases
+        )
+        exclusion_filter_bases = [
+            ["--exclude", f"{build_cache_resolved}/{exclusion}"]
+            for exclusion in exclusion_filter_bases
+        ]
+
+        build_cache_path = (
+            (
+                builder.build_dir
+                if _using_root(builder, context, self.scope)
+                else builder.get_build_cache_path(context)
+            )
+            .absolute()
+            .resolve()
+        )
+
+        coverage_output_dir = context.absolute().resolve() / "coverage"
         coverage_output_dir.mkdir(exist_ok=True)
-        project_root = builder.get_settings(
-            "project_root",
-            builder.get_settings("framework_path", builder.build_dir.parent.parent),
+        project_root = (
+            builder.get_settings(
+                "project_root",
+                builder.get_settings("framework_path", builder.build_dir.parent.parent),
+            )
+            .absolute()
+            .resolve()
         )
         filter_path = (
-            Path(project_root).absolute()
-            if _using_root(builder, context, self.scope)
-            else _get_project_path(builder, context)
+            (
+                Path(project_root).absolute()
+                if _using_root(builder, context, self.scope)
+                else _get_project_path(builder, context)
+            )
+            .absolute()
```

Comment:
From [the docs on resolve()](https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve) : 
> **Path.resolve:** Make the path absolute, resolving any symlinks

So .absolute() would be redundant here? And it seems like there were issues/debates around .absolute() prior to Python 3.11 (see [here](https://bugs.python.org/issue29688)) so it might be wise to get rid of it.